gufe/tests/test_proteincomponent.py

```diff
@@ -239,6 +239,21 @@ def test_to_openmm_topology(self, in_pdb_path):
         assert openmm_top.getNumChains() == gufe_openmm_top.getNumChains()
         assert openmm_top.getNumResidues() == gufe_openmm_top.getNumResidues()
         assert openmm_top.getNumChains() == gufe_openmm_top.getNumChains()
+        # Make sure bond.order is the expected type int or None
+        assert all(isinstance(bond.order, (int, type(None))) for bond in openmm_top.bonds())
+
+    @pytest.mark.parametrize("in_pdb_path", ALL_PDB_LOADERS.keys())
+    def test_protein_component_openmm_bond(self, in_pdb_path):
+        """
+        Test that `to_openmm_topology().bonds()` produces a valid bond object.
+        """
+        in_pdb_io = ALL_PDB_LOADERS[in_pdb_path]()
+        prot = self.cls.from_pdb_file(in_pdb_io, name="Alice")
+        omm_topology_bonds = prot.to_openmm_topology().bonds()
+
+        with pytest.raises(StopIteration):
+            while True:
+                repr(next(omm_topology_bonds))
```

Comment:
Is there a reason we don't iterate over the iterator and just call `repr` on each object? While this works, it seems odd that do it like this. I think if you do have a good reason for doing it like, I would instead use a `try`/`except` block here, since I tend to use `pytest.raises` when I am testing to see if my code raises an exception that I expect/want to test. Here we just don't care if we exhaust the iterator.

Reply:
No special reason other than not being able to be 100% sure that there's always going to be a `TypeError` exception when we use the `Bond.__repr__` method from OpenMM. That is, if I'm understanding you, if we try to catch the `TypeError` exception, there's no guarantee that the `repr(bond)` wouldn't raise any other exception that we would just not catch. Does this make sense?

I agree this is commonly not the way we expect `pytest.raises` to be used, but in this case we are using OpenMM's `Bond.__repr__` as a validator and I want to test for success rather than failure, and `StopIteration` is kind of a success in this case. I may be overthinking this, but I accept suggestions.